### PR TITLE
Add sanity tests for HCO alerts

### PR DESCRIPTION
**Pull request by assafad**
This PR adds new test cases for HCO alert definitions. We expect to see `summary`, `severity`, `kubernetes_operator_part_of` and `kubernetes_operator_component` for each alert. Those fields must not be empty and some must have specific values.

Signed-off-by: assafad <aadmi@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
None
```

### CODE REVIEW

File: tests/func-tests/monitoring_test.go [hunk 1]
```diff
@@ -82,22 +104,39 @@ func getAlertByName(alerts promApiv1.AlertsResult, alertName string) *promApiv1.
...
+	url, ok := rule.Annotations["runbook_url"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have runbook_url annotation", rule.Alert))
+	resp, err := runbookClient.Head(url)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("%s runbook is not available", rule.Alert))
+	ExpectWithOffset(1, resp.StatusCode).Should(Equal(200), fmt.Sprintf("%s runbook is not available", rule.Alert))
+}
+
+func checkSummary(rule monitoringv1.Rule) {
+	summary, ok := rule.Annotations["summary"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have summary annotation", rule.Alert))
```
Comment (nunnatsa) [inline tests/func-tests/monitoring_test.go:117]: Please use built-in matcher; e.g.

```suggestion
	ExpectWithOffset(1, rule.Annotations).Should(HaveKey("summary"), fmt.Sprintf("%s does not have summary annotation", rule.Alert))
	ExpectWithOffset(1, rule.Annotations["summary"]).ShouldNot(BeEmpty(), fmt.Sprintf("%s has an empty summary", rule.Alert)
```

File: tests/func-tests/monitoring_test.go [hunk 2]
```diff
@@ -82,22 +104,39 @@ func getAlertByName(alerts promApiv1.AlertsResult, alertName string) *promApiv1.
...
+	ExpectWithOffset(1, summary).ToNot(BeEmpty(), fmt.Sprintf("%s has an empty summary", rule.Alert))
+}
+
+func checkSeverityLabel(rule monitoringv1.Rule) {
+	severity, ok := rule.Labels["severity"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have severity label", rule.Alert))
+	ExpectWithOffset(1, severity).To(BeElementOf("info", "warning", "critical"), fmt.Sprintf("%s severity label is not valid", rule.Alert))
+}
```
Comment (nunnatsa) [inline tests/func-tests/monitoring_test.go:125]: Please use built-in matchers; e.g.

```suggestion
ExpectWithOffset(1, rule.Labels).Should(HaveKeyWithValue("severity", BeElementOf("info", "warning", "critical")), fmt.Sprintf("%s severity label is missing or invalid", rule.Alert)))
```

File: tests/func-tests/monitoring_test.go [hunk 3]
```diff
@@ -82,22 +104,39 @@ func getAlertByName(alerts promApiv1.AlertsResult, alertName string) *promApiv1.
...
+	kubernetesOperatorPartOf, ok := rule.Labels["kubernetes_operator_part_of"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have kubernetes_operator_part_of label", rule.Alert))
+	ExpectWithOffset(1, kubernetesOperatorPartOf).To(Equal("kubevirt"), fmt.Sprintf("%s kubernetes_operator_part_of label is not valid", rule.Alert))
+}
+
+func checkComponentLabel(rule monitoringv1.Rule) {
+	kubernetesOperatorComponent, ok := rule.Labels["kubernetes_operator_component"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have kubernetes_operator_component label", rule.Alert))
+	ExpectWithOffset(1, kubernetesOperatorComponent).To(Equal("hyperconverged-cluster-operator"), fmt.Sprintf("%s kubernetes_operator_component label is not valid", rule.Alert))
 }
```
Comment (nunnatsa) [inline tests/func-tests/monitoring_test.go:137]:
```suggestion
ExpectWithOffset(1, rule.Labels).Should(HaveKeyWithValue("kubernetes_operator_component", "hyperconverged-cluster-operator"), fmt.Sprintf("%s kubernetes_operator_component label is missing or invalid", rule.Alert))
```

File: tests/func-tests/monitoring_test.go (hunk 1 above)
Comment (nunnatsa) [reply tests/func-tests/monitoring_test.go:117]: This *may* will ***maybe*** also work:

```suggestion
	ExpectWithOffset(1, rule.Annotations).Should(HaveKeyWithValue("summary", Not(BeEmpty()), fmt.Sprintf("%s summery is missing or empty", rule.Alert))
```

File: tests/func-tests/monitoring_test.go [hunk 4]
```diff
@@ -45,11 +46,32 @@ var _ = Describe("[crit:high][vendor:cnv-qe@redhat.com][level:system]Monitoring"
...
+					checkSummary(rule)
+				}
+			}
+		}
+	})
+
+	It("Alert rules should have all the requried labels", func() {
+		for _, group := range prometheusRule.Spec.Groups {
+			for _, rule := range group.Rules {
+				if len(rule.Alert) > 0 {
```
[the remark this answers is not in the document]
Comment (assafad) [reply tests/func-tests/monitoring_test.go]: In these tests we shouldn't. We have recording rules, which use `Record` and `Expr`, and their `Alert` is empty. This way we overlook them, and only check alerts.

**Review by nunnatsa — APPROVED** [review]
/approve

**Pull request merged**